File: knowledge/algorithms/leetcode/00001TwoSum/TwoSum.js


function TwoSum(num_lst, target, num=2){
    let numTwoIndex = new Map()   
    for (let i =0; i <num_lst.length;i++){
        let complement = target - num_lst[i];
        if (numTwoIndex.has(complement)){
            return [numTwoIndex.get(complement), i]
        }
        numTwoIndex.set(num_lst[i],i)
    }
}
console.log(TwoSum([2,7,11,15], 9))

File: knowledge/algorithms/leetcode/00001TwoSum/TwoSum.py


# def TwoSum(num_lst, target, nums=2):
#     for i, x in enumerate(num_lst):
#         for j, y in enumerate(num_lst):
#             if i != j and (x+y) == target:
#                 a = (i,j)
#                 break 
#         break    
#     return a       

def TwoSum(num_lst, target, nums=2):
    num_to_idx = {}
    for i, num in enumerate(num_lst):
        complement = target - num
        if complement in num_to_idx:
            return [num_to_idx[complement], i]
        num_to_idx[num] = i


print(TwoSum([2,7,11,15], 9))
            
